### PR TITLE
Collate 1d updates

### PR DESCRIPTION
**Pull request by badpandabear**
Updates to pypeit_collate_1d for ADAP 2019. Changes include:

- Rejecting a SpecObj if it is entirely masked out
- Default collate 1d to not checking the datamodel version when loading files.
- Adding a chk_version flag to force checking the datamodel version if desired.

### CODE REVIEW

File: pypeit/core/collate.py [hunk 1]
```diff
@@ -68,23 +68,25 @@ def __init__(self, spec1d_obj, spec1d_header, spec1d_file, spectrograph, match_t
...
 
     @classmethod
-    def build_source_objects(cls, spec1d_files, match_type):
+    def build_source_objects(cls, specobjs_list, spec1d_files, match_type):
         """Build a list of SourceObjects from a list of spec1d files. There will be one SourceObject per
         SpecObj in the resulting list (i.e. no combining or collating is done by this method).
 
         Args:
-            spec1d_files (list of str): List of spec1d filenames
+            spec1d_files (list of :obj:`pypeit.specobjs.SpecObjs`): List of SpecObjs objects to build from.
```
Comment (profxj) [inline pypeit/core/collate.py]: `specobjs_list` ?

**Review by profxj — APPROVED** [review]
Looks good;  one minor edit from me.

File: pypeit/scripts/collate_1d.py [hunk 2]
```diff
@@ -160,9 +160,7 @@ def exclude_source_objects(source_objects, exclude_map, par):
               :class:`~pypeit.core.collate.SourceObject` with any excluded ones
               removed.
 
-            - **missing_archive_msgs** (:obj:`list`): A list of messages
-              explaining why some source objects were excluded.
-
+                **excluded_messages** (:obj:`list`): A list of messages explaining why some source objects were excluded.
```
Comment (debora-pe) [inline pypeit/scripts/collate_1d.py]: I think here it's missing the `-` for the bullet list

**Review by debora-pe — APPROVED** [review]
Looks good. Thank you @badpandabear!

**Comment by badpandabear** [comment]
dev-suite run:
```
Test Summary
--------------------------------------------------------
--- PYTEST PYPEIT UNIT TESTS PASSED  232 passed, 73 warnings in 370.06s (0:06:10) ---
--- PYTEST UNIT TESTS FAILED  5 failed, 118 passed, 191 warnings in 808.21s (0:13:28) ---
--- PYTEST VET TESTS PASSED  47 passed, 29 warnings in 636.68s (0:10:36) ---
--- PYPEIT DEVELOPMENT SUITE PASSED 207/207 TESTS  ---
Testing Started at 2023-08-10T02:58:47.575034
Testing Completed at 2023-08-10T10:54:18.445617
Total Time: 7:55:30.870583
```

The failed unit tests were later fixed by 
https://github.com/pypeit/PypeIt-development-suite/pull/272

**Pull request merged**